src/functions/schema-path.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Schema Path is a variant of the normal Schema rule with a slight twist.
+ * Instead of passing the rule a schema to validate, there is already a schema
+ * somewhere inside the target, and we are just selecting it, and applying it to
+ * a given field.
+ *
+ * The primary use case for this was validating OpenAPI examples
+ * against their schema, but this could be used for other things.
+ */
+import { IFunction, ISchemaPathOptions } from '../types';
+import { schema } from './schema';
+
+import jp = require('jsonpath');
+
+export const schemaPath: IFunction<ISchemaPathOptions> = (targetVal, opts, paths, otherValues) => {
+  if (!targetVal) return [];
+  let value = targetVal;
+
+  // We want to run this on a specific field, for example: example
+  if (opts.field && typeof targetVal === 'object') {
+    value = targetVal[opts.field];
+  }
+
+  if (!value) return [];
+
+  // Hunt in the original targetVal for the schema we are going to apply
+  let schemaObj;
+  try {
+    schemaObj = jp.value(targetVal, opts.schemaPath);
```

Comment:
Should we remove the field value we are testing on out of the `schemaObj`?

So if we are testing the `example` property in a schema should we remove that property from the schema

Reply:
`delete schemaObj[opts.field]`

Reply:
It doesnt hurt having it there. It's being ignored by the validator, which is why we need this rule. :)